packages/manager/apps/web-hosting/src/pages/dashboard/AddDomain.page.tsx

```diff
@@ -0,0 +1,721 @@
+import React, { useEffect, useState } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { useTranslation, Trans } from 'react-i18next';
+import { aapi } from '@ovh-ux/manager-core-api';
+
+import {
+  OdsButton,
+  OdsCard,
+  OdsModal,
+  OdsInput,
+  OdsRadio,
+  OdsText,
+  OdsCheckbox,
+  OdsFormField,
+  OdsSelect,
+  OdsTooltip,
+  OdsMessage,
+  OdsIcon,
+} from '@ovhcloud/ods-components/react';
+import {
+  ODS_BUTTON_COLOR,
+  ODS_BUTTON_VARIANT,
+  ODS_INPUT_TYPE,
+  ODS_MESSAGE_COLOR,
+  ODS_MODAL_COLOR,
+  ODS_TEXT_PRESET,
+} from '@ovhcloud/ods-components';
+import { useNotifications } from '@ovh-ux/manager-react-components';
+
+import {
+  useCreateAttachedDomainsService,
+  useGetDomainZone,
+  useGetHostingService,
+} from '@/data/hooks/webHostingDashboard/useWebHostingDashboard';
+import { isValidDomain } from '@/utils/validator';
+import {
+  HostingCountries,
+  HostingDomainStatus,
+  TExistingDomain,
+} from '@/data/type';
+import { subRoutes, urls } from '@/routes/routes.constants';
+
+interface FormState {
+  path: 'A' | 'B';
+  domain?: string;
+  subDomain: string;
+  wwwNeeded: boolean;
+  folder: string;
+  ip?: boolean;
+  selectedIp?: HostingCountries;
+  firewall?: boolean;
+  logs?: boolean;
+  domainLogs?: string;
+  automaticConfiguration?: boolean;
+}
+const initialState: FormState = {
+  path: 'A',
+  subDomain: '',
+  folder: '',
+  wwwNeeded: true,
+  automaticConfiguration: true,
+};
+
+export default function SectigoModal() {
+  const { serviceName } = useParams();
+  const navigate = useNavigate();
+  const closeModal = () =>
+    navigate(urls.ssl.replace(subRoutes.serviceName, serviceName));
+  const { data } = useGetHostingService(serviceName);
+  const domainZone = useGetDomainZone();
+  const { addSuccess, addWarning } = useNotifications();
+  const { t } = useTranslation('dashboard');
+  const [step, setStep] = useState<number | null>(0);
+  const [form, setForm] = useState<FormState>(initialState);
```

Comment:
the recommended way of handling forms is to use `react-hook-form` and `zod` is there any reason for you to not use it ?